server/capabilities/gke_networks_endpoint.go

```diff
@@ -0,0 +1,89 @@
+package capabilities
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"google.golang.org/api/compute/v1"
+	"net/http"
+)
+
+// NewGKENetworksHandler creates a new GKENetworksHandler
+func NewGKENetworksHandler() *GKENetworksHandler {
+	return &GKENetworksHandler{}
+}
+
+// GKENetworksHandler lists available networks in GKE
+type GKENetworksHandler struct {
+}
+
+func (g *GKENetworksHandler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		writer.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	writer.Header().Set("Content-Type", "application/json")
+
+	var body zoneCapabilitiesRequestBody
+	err := extractRequestBody(writer, req, &body)
+
+	if err != nil {
+		handleErr(writer, err)
+		return
+	}
+
+	err = validateCapabilitiesRequestBody(writer, &body.capabilitiesRequestBody)
+
+	if err != nil {
+		handleErr(writer, err)
+		return
+	}
+
+	if body.Credentials == "" {
+		writer.WriteHeader(http.StatusBadRequest)
+		handleErr(writer, fmt.Errorf("invalid credentials"))
+		return
+	}
+
+	client, err := g.getServiceClient(context.Background(), body.Credentials)
+
+	if err != nil {
+		writer.WriteHeader(http.StatusInternalServerError)
+		handleErr(writer, err)
+		return
+	}
```

Comment:
I see a duplication of code between all the GKE request up until about this point, can we extract that out into some common code for the various endpoints?

Reply:
Yep, this question maybe needs a separate issue, so I will not update it in this PR.

Reply:
If you don't want to refactor existing funcs I'm fine with that, but you are adding a lot of duplicated code in this PR and I don't want to LGTM a PR like that unless we have a compelling reason to.

Reply:
should be addressed in this pr

Reply:
OK, I will address it in this pr and as for those existing code duplication, I will add a separate commit.